src/utils/createComponent.ts

```diff
@@ -98,4 +98,18 @@ export function createComponent<I extends HTMLElement, E extends EventNames = {}
         }
       : options,
   );
+
+  return React.forwardRef<typeof component, Parameters<typeof component>[0]>((props, ref) => {
+    // Map falsy boolean properties as `undefined` to avoid them from rendering with the
+    // value "false" in the attribute, for example `<vaadin-button hidden="false">`,
+    // which would actually evaluate as `hidden` being `true`.
+    const falsyBooleanProps = Object.entries(props).reduce((acc, [key, value]) => {
+      if (value === false) {
+        return { ...acc, [key]: undefined };
```

Comment:
We probably could use `null` for falsy values to align with the logic implemented in https://github.com/facebook/react/pull/24541 (expected to land in React 19, currently still experimental feature).

Reply:
No longer relevant since https://github.com/vaadin/react-components/pull/171#discussion_r1425398926